### PR TITLE
Remove 0.5*timestep logic from call to zm

### PR DESCRIPTION
**Pull request by cacraigucar**
ZM has for a very long time passed in .5*timestep and then inside multiplied by 2 in several places (and also done 0.5/input_timestep in several other locations).  It also hinders CCPP'ization as we do not have a half timestep standard name.  This confusing, unnecessary logic will be removed.

To confirm the results, see this email from @adamrher

Your ZM branch has round-off level differences with cam6_4_018, so it's working as expected (plots attached).
Ignore the top right panel "cam6_4_015" -- that's my clubb+mf branch and something is very wrong with that run.

### CODE REVIEW

**Comment by cacraigucar** [comment]
[temp_PRECC.pdf](https://github.com/user-attachments/files/16626896/temp_PRECC.pdf)

File: zhang_mcfarlane/zm_conv_convtran.F90 [hunk 1]
```diff
@@ -68,9 +68,6 @@ subroutine zm_conv_convtran_run(ncol, pver, &
 
    real(kind_phys), intent(in) :: dpdry(:,:)    ! Delta pressure between interfaces        (ncol,pver)
 
-   real(kind_phys), intent(in) :: dt                      ! 2 delta t (model time increment)
```
Comment (adamrher) [inline zhang_mcfarlane/zm_conv_convtran.F90:71]: I presume dt is just not used in convtran?

**Review by adamrher — APPROVED** [review]
Looks good to me!

File: zhang_mcfarlane/zm_conv_convtran.F90 (hunk 1 above)
Comment (cacraigucar) [reply zhang_mcfarlane/zm_conv_convtran.F90:71]: That is correct - when I went to see if the dt needed adjusting in this routine, I discovered it was not used, so I removed it.

File: zhang_mcfarlane/zm_conv_momtran.F90 [hunk 2]
```diff
@@ -59,7 +59,7 @@ subroutine zm_conv_momtran_run(ncol, pver, pverp, &
    real(kind_phys), intent(in) :: ed(:,:)       ! Mass entraining from downdraft            (ncol,pver)
    real(kind_phys), intent(in) :: dp(:,:)       ! Delta pressure between interfaces         (ncol,pver)
    real(kind_phys), intent(in) :: dsubcld(:)       ! Delta pressure from cloud base to sfc  (ncol)
-   real(kind_phys), intent(in) :: dt                   !  time step in seconds : 2*delta_t
+   real(kind_phys), intent(in) :: dt
```
Comment (peverwhee) [inline zhang_mcfarlane/zm_conv_momtran.F90:62]: can we put the `!  time step in seconds` comment back?

**Review by peverwhee — APPROVED** [review]
one tiny thing!

File: zhang_mcfarlane/zm_conv_momtran.F90 (hunk 2 above)
Comment (adamrher) [reply zhang_mcfarlane/zm_conv_momtran.F90]: That's fine by me @cacraigucar

**Pull request merged**